### PR TITLE
Fix for not sending joint steps when diff is small

### PR DESCRIPTION
**Pull request by spieswl**
@biotinker is OOO today, but this was a high priority problem @mestcihazal ran into when working with xArms and I have been able to recreate and fix on my end.

Pretty small fix overall so @biotinker is included primarily for awareness. If you approve, @randhid, we can just get this in and I'll alert DevRel of the fix.

### CODE REVIEW

File: components/arm/xarm/xarm_comm.go [hunk 1]
```diff
@@ -357,9 +357,9 @@ func (x *xArm) MoveToJointPositions(ctx context.Context, newPositions *pb.JointP
 
 	diff := getMaxDiff(from, to)
 	nSteps := int((diff / float64(x.speed)) * x.moveHZ)
```
Comment (spieswl) [inline components/arm/xarm/xarm_comm.go:359]: Original problem is actually here; if diff was small enough, the cast to an `int` would result in an `nSteps` of 0 which skipped the entire loop where we issued `MoveJoints` commands.

**Review by randhid — APPROVED** [review]
LGTM, nice catch.  I don't see anything that could be hardware-breaking. But will keep an eye out.

File: components/arm/xarm/xarm_comm.go [hunk 2]
```diff
@@ -374,14 +374,33 @@ func (x *xArm) MoveToJointPositions(ctx context.Context, newPositions *pb.JointP
...
 		}
 		if !utils.SelectContextOrWait(ctx, time.Duration(1000000./x.moveHZ)*time.Microsecond) {
 			return ctx.Err()
 		}
+		return nil
 	}
+
+	// every step except the last, skipped if diff is small enough
+	for i := 1; i < nSteps; i++ {
+		step := referenceframe.InputsToFloats(referenceframe.InterpolateInputs(from, to, float64(i)/float64(nSteps)))
```
[the remark this answers is not in the document]
Comment (spieswl) [reply components/arm/xarm/xarm_comm.go:390]: Yeah, this doesn't solve the divide-by-zero issue, but I will add and accept a quick suggestion to improve the comment here.

File: components/arm/xarm/xarm_comm.go [hunk 3]
```diff
@@ -374,14 +374,33 @@ func (x *xArm) MoveToJointPositions(ctx context.Context, newPositions *pb.JointP
...
 		if err != nil {
 			return err
 		}
 		if !utils.SelectContextOrWait(ctx, time.Duration(1000000./x.moveHZ)*time.Microsecond) {
 			return ctx.Err()
 		}
+		return nil
 	}
+
+	// every step except the last, skipped if diff is small enough
```
Comment (spieswl) [inline components/arm/xarm/xarm_comm.go:388]:
```suggestion
	// every step except the last, skipped if diff is small enough. Note that if diff calculations are small enough, nSteps could be zero, leading to a bad situation inside the loop
```

File: components/arm/xarm/xarm_comm.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (biotinker) [reply components/arm/xarm/xarm_comm.go:391]: We can't actually ever divide by zero here.

If nSteps is zero, we never enter the loop.

[the remark this answers is not in the document]
Comment (randhid) [reply components/arm/xarm/xarm_comm.go:391]: We know, we just don't like that there is no explicit check.

**Pull request merged**